src/inject.js

```diff
@@ -79,10 +79,18 @@ async function handleRequest (req) {
   if(req.method === 'eth_requestAccounts') {
     return await window[injectionName].enable('${chain}')
   }
-  if(req.method === 'personal_sign') { 
+  if(req.method === 'personal_sign') {
     const sig = await eth.getMethod('wallet.signMessage')(req.params[0], req.params[1])
     return '0x' + sig
   }
+
+  if(req.method === 'eth_signTypedData' ||
+    req.method === 'eth_signTypedData_v3' ||
+    req.method === 'eth_signTypedData_v4') {
+    const sig = await eth.getMethod('wallet.signTypedMessage')(req)
```

Comment:
Shouldn't this be added to the list of `CONFIRM_REQUIRED` requests???
I also see no UI changes, but this type of action shouldn't be automatically signed by the wallet and should have a confirm prompt.